Firestore/Swift/Source/Codable/third_party/FirestoreDecoder.swift

```diff
@@ -978,6 +978,9 @@ extension _FirestoreDecoder {
 
   func unbox(_ value: Any, as type: Date.Type) throws -> Date? {
     guard !(value is NSNull) else { return nil }
+    if let timestamp = value as? Timestamp {
```

Comment:
Maybe add a comment like"
`Firestore returns all dates as Timestamp, converting it to Date so it can be used in custom objects.`.

Reply:
Done!